docs: update Documentation to show single Document serving (#6009)
jina-ai/jina: docs/concepts/preliminaries/index.md

    Document is the basic object for representing multimodal data. It can be extended to represent any data you want. More information can be found in [DocArray's Docs](https://docs.docarray.org/user_guide/representing/first_step/). 

- **DocList**
    DocList is a list-like container of multiple Documents. More information can be found in [DocArray's Docs](https://docs.docarray.org/user_guide/representing/array/).

> All the components in Jina use BaseDoc and/or DocList as the main data format for communication, making use of the different 

Reviewer: ```suggestion
All the components in Jina use `BaseDoc` and/or `DocList` as the main data format for communication, making use of the different 
```

> serialization capabilities of these structure.

Reviewer: ```suggestion
serialization capabilities of these structures.
```

 

- **Executor**

>     {class}`~jina.Executor` is a Python class that can serve logic using Documents. Loosely speaking, each Executor is a service wrapping a model or application.

Reviewer: ```suggestion
    A {class}`~jina.Executor` is a Python class that serves logic using Documents. Loosely speaking, each Executor is a service wrapping a model or application.
```

- **Gateway**

>     Gateway is the entrypoint of a {term}`Flow`. It exposes multiple protocols for external communications; it routes all internal traffic to different Executors that collaborately 

Reviewer: ```suggestion
    A Gateway is the entrypoint of a {term}`Flow`. It exposes multiple protocols for external communications; routing all internal traffic to different Executors that work together to 
```

    {class}`~jina.Flow` ties multiple {class}`~jina.Deployments`s together into a logic pipeline to achieve a more complex task. It orchestrates both {term}`Executor`s and the {term}`Gateway`.

**Client**

> {class}`~jina.Client` connects to a {term}`Gateway` or {term}`Executor` and sends/receives/streams data from them.

Reviewer: ```suggestion
The {class}`~jina.Client` connects to a {term}`Gateway` or {term}`Executor` and sends/receives/streams data from them.
```
